### ISSUE TITLE
Maybe rename private messages?

### ISSUE THREAD

**Issue by stasm** [projectfluent/fluent#85]
We've seen some confusion about how to talk about private messages and their attributes. The message itself is not really private -- tools will still check if it exists. What's private is its interface: the value and any attributes.

Perhaps we could find a better wording to make this easier to understand? in #62 I originally proposed _glossary messages`; the feedback was that in the world of localization _glossaries_ already have a defined meaning and it's better not to confuse people further.

**Comment by stasm** [comment]
How about _vocabulary terms_?

We'd then have _messages_ with _public attributes_ and _vocabulary terms_ with _private messages_.

**Comment by flodolo** [comment]
I find _vocabulary_  as confusing as _glossary_ for these messages:

> A vocabulary is a set of familiar words within a person's language. 

Also, calling one *attribute* and the other *message* is not helping. They're both messages, because they're defined the same way. Maybe *internal messages* with *private attributes*? In the end, I believe any definition we come up with will need an explanation, and won't be intuitive.

**Comment by stasm** [comment]
> Also, calling one attribute and the other message is not helping.

That's not what I meant; I must have not expressed it clearly. Let me try again:

```properties
# This is a vocabulary term with an attribute. The attribute is private.
-brand-name = Firefox
    .gender = masculine

# This is a message with an attribute. The attribute is public.
about-app = About { -brand-name }
    .title = About this application
```

**Comment by stasm** [comment]
My point is that `-brand-name` is not really a _message_—you're not supposed to put an entire sentence in it and it doesn't show up in the UI on its own. It's a single word or term to be used in other messages.

**Comment by stasm** [comment]
Or maybe simply _terms_ would work? You'd define _terms_ which can be used in _messages_.

**Comment by stasm** [comment]
We could also consider calling the _expressions_ although we already use this word in a very specific meaning to describe contents of placeables which evaluate to something.

**Comment by flodolo** [comment]
> Or maybe simply terms would work? You'd define terms which can be used in messages.

That would make sense to me: `terms` have private attributes, `messages` have public attributes. 

With the note that, while we could give guidelines on how to use them, we can't really stop people from using an entire sentence.

**Comment by stasm** [comment]
> With the note that, while we could give guidelines on how to use them, we can't really stop people from using an entire sentence.

Right. Which is also why I suggested _terms_ rather than _words_ for example. Just in case :)

**Comment by stasm** [comment]
I would like to reach an agreement on this today and make this part of the Syntax 0.5 spec.

**Comment by Pike** [comment]
I've read up on [access specifiers](https://en.wikipedia.org/wiki/Access_modifiers) to get some 3rd party definitions of _private_.

I don't think that the term (pun) is missing the point, the private messages are only accessible from within the MessageContext, but not outside of it. I don't think that they're treatment in static analysis (compare-locales) should impact that.

So I think that we should continue to use the word _private_ to describe the concept. _protected_ might be a tad closer, but that's also an order of magnitude more technical, so I'd rather not go fo r that.

That said.

> The difference between Private Messages and Messages is that Messages are public and have public atrributes, but Private Messages ....

What a mouthful. That's why I think _Term_ is actually a good idea.

> The difference between Messages and Terms are that Messages and their attributes are exposed to the program, while Terms are private to the context, and their attributes can vary from localization to localization.

Sounds easier to me.

**Comment by stasm** [comment]
Great points, @Pike, thanks! I'll prepare a PR.
